pkg/cloud-provider/cloudapi/azure/azure_api_wrappers.go

```diff
@@ -16,236 +16,264 @@ package azure
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"net/http"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-03-01/network"
-	"github.com/Azure/azure-sdk-for-go/services/resourcegraph/mgmt/2021-03-01/resourcegraph"
-	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	resourcegraph "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resourcegraph/armresourcegraph"
 )
 
 type azureNwIntfWrapper interface {
 	createOrUpdate(ctx context.Context, resourceGroupName string, networkInterfaceName string,
-		parameters network.Interface) (network.Interface, error)
-	listAllComplete(ctx context.Context) ([]network.Interface, error)
+		parameters armnetwork.Interface) (armnetwork.Interface, error)
+	listAllComplete(ctx context.Context) ([]armnetwork.Interface, error)
 }
 type azureNwIntfWrapperImpl struct {
-	nwIntfAPIClient network.InterfacesClient
+	nwIntfAPIClient armnetwork.InterfacesClient
 }
 
 func (nwIntf *azureNwIntfWrapperImpl) createOrUpdate(ctx context.Context, resourceGroupName string, networkIntfName string,
-	parameters network.Interface) (network.Interface, error) {
-	var nwInterface network.Interface
+	parameters armnetwork.Interface) (armnetwork.Interface, error) {
+	var nwInterface armnetwork.Interface
 	nwIntfClient := nwIntf.nwIntfAPIClient
-	future, err := nwIntfClient.CreateOrUpdate(ctx, resourceGroupName, networkIntfName, parameters)
+	poller, err := nwIntfClient.BeginCreateOrUpdate(ctx, resourceGroupName, networkIntfName, parameters, nil)
```

Comment:
BeginCreateOrUpdate returns a poller object. On this object we can perform a bunch of operations. 
So poller seems to be appropriate (rather renaming it to op)


type Poller[T any] struct {
	op     PollingHandler[T]
	resp   *http.Response
	err    error
	result *T
	done   bool
}